tests/components/cert_expiry/test_init.py

```diff
@@ -107,3 +108,26 @@ async def test_unload_config_entry(mock_now, hass):
     await hass.async_block_till_done()
     state = hass.states.get("sensor.cert_expiry_timestamp_example_com")
     assert state is None
+
+
+async def test_delay_load_during_startup(hass):
+    """Test loading a new config entry during runtime."""
+    hass.state = CoreState.starting
+
+    entry = MockConfigEntry(domain=DOMAIN, data={CONF_HOST: HOST, CONF_PORT: PORT})
+    entry.add_to_hass(hass)
+
+    timestamp = future_timestamp(100)
+    with patch(
+        "homeassistant.components.cert_expiry.get_cert_expiry_timestamp",
+        return_value=timestamp,
+    ):
+        assert await async_setup_component(hass, DOMAIN, {}) is True
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
```

Comment:
Don't we want to test setting up the integration when the core is already started?

Reply:
All of the existing tests are run when core is already fully started, this test is there to validate it works when loading before core startup is complete.

Is it worth asserting `hass.state is CoreState.running` in the other tests for clarity?

Reply:
Yes, and change the docstring of this test.